source/posix/thread.c

```diff
@@ -296,12 +296,11 @@ int aws_thread_launch(
             attr_return = pthread_attr_setaffinity_np(attributes_ptr, sizeof(cpuset), &cpuset);
 
             if (attr_return) {
-                AWS_LOGF_ERROR(
+                AWS_LOGF_WARN(
                     AWS_LS_COMMON_THREAD,
-                    "id=%p: pthread_attr_setaffinity_np() failed with %d.",
+                    "id=%p: pthread_attr_setaffinity_np() failed with %d. Continuing without cpu affinity",
                     (void *)thread,
                     attr_return);
-                goto cleanup;
```

Comment:
I don't like we just continue here:
1. if anything fails from `pthread_create`, we still check if we have the cpu_id set, and we will retry, which will lead to the same `pthread_create` and probably fail the same way.
2. In my opinion, it's easier to follow if we just `goto cleanup` here.

Reply:
Thanks, I have re-added the goto cleanup line.